parsers/ocaml.c

```diff
@@ -1730,12 +1929,22 @@ static void localScope (vString * const ident, ocaToken what)
 		toDoNext = &contextualTillToken;
 		waitedToken = Tok_CurlR;
 		comeAfter = &localScope;
-		contextualTillToken (ident, what);
+		contextualTillToken (ident, what, whatNext);
 		break;
 
 		/* Yeah imperative feature of OCaml,
 		 * a ';' like in C */
 	case Tok_semi:
+		/* ';;' case should end all scopes */
```

Comment:
Probably? I don't actually remember why I didn't do that. The OCaml lexer treats `;;` as a special token, so it would make sense to do the same here.